include/game/bases/d_CourseSelectGuide.hpp

```diff
@@ -0,0 +1,288 @@
+    /// @brief The state manager for the button legends during the map view.
+    sFStateMgr_c<dCourseSelectGuide_c, sStateMethodUsr_FI_c> mStateMgrScrollGuide;
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, WorldCourseOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, WorldCourseOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, WorldCourseDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, WorldCourseExitAnimeEndCheck);
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, GuideOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, GuideOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, GuideDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, GuideExitAnimeEndCheck);
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, UpOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, UpOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, UpDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, UpExitAnimeEndCheck);
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, DownOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, DownOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, DownDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, DownExitAnimeEndCheck);
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, LeftOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, LeftOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, LeftDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, LeftExitAnimeEndCheck);
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, RightOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, RightOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, RightDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, RightExitAnimeEndCheck);
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ShadowOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ShadowOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ShadowDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ShadowExitAnimeEndCheck);
+
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ScrollGuideOnStageWait);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ScrollGuideOnStageAnimeEndCheck);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ScrollGuideDisp);
+    STATE_FUNC_DECLARE(dCourseSelectGuide_c, ScrollGuideExitAnimeEndCheck);
+
+    nw4r::lyt::Pane *mpRootPane; ///< The root pane of the view.
+
+    nw4r::lyt::Pane *mpNullPanes[N_COUNT]; ///< The null panes of the view.
+    LytTextBox_c *mpTextBoxes[T_COUNT]; ///< The textboxes of the view.
+    nw4r::lyt::Picture *mpPicturePanes[P_COUNT]; ///< The picture panes of the view.
+
+    Remocon::EXTENSION_TYPE_e mExtension; ///< The currently attached Wii remote extension.
+    int mWorldNo; ///< The current world number.
+    int mCourseNo; ///< The current course number. Set to a negative value if it's a course with an icon.
+    int mCourseType; ///< The current course type. @todo Document this enum.
```

Comment:
The values kind of document themselves from CourseSelectSet(), don't think it's much work to add this enum. However, i'm not sure if it's also used outside of this class.